cowprotocol/accounting/fees/classified_fees_4058574.sql

```diff
@@ -0,0 +1,60 @@
+-- This query computes fees
+--
+-- It currently uses the raw_order_rewards table which is only available on ethereum. On other chains it returns an empty table.
+--
+-- Parameters:
+--  {{start_time}} - the timestamp for which the analysis should start (inclusively)
+--  {{end_time}} - the timestamp for which the analysis should end (exclusively)
+--  {{blockchain}} - network to run the analysis on
+--
+-- The columns of the result are
+-- - block_time: time of settlement transaction
+-- - tx_hash: settlement transaction hash
+-- - token_address: address of token with a balance change. contract address for erc20 tokens,
+--   0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee for native token
+-- - amount: value of fee in atoms of the token
+-- - transfer_type: 'protocol_fee' for the total protocol fee (including partner fee), 'network_fee' for network fees
+
+with fee_data as (
+    select
+        block_time,
+        t.tx_hash,
+        t.order_uid,
+        atoms_sold,
+        atoms_bought,
+        sell_token_address,
+        cast(cast(data.protocol_fee as varchar) as int256) as protocol_fee, -- noqa: RF01
+        cast(cast(data.surplus_fee as varchar) as int256) as surplus_fee, -- noqa: RF01
+        from_hex(data.protocol_fee_token) as protocol_fee_token_address -- noqa: RF01
+    from cow_protocol_{{blockchain}}.trades as t
+    inner join cowswap.raw_order_rewards as ror -- this table only exists for ethereum trades
+        on t.order_uid = from_hex(ror.order_uid) and t.tx_hash = from_hex(ror.tx_hash)
+    where block_time >= cast('{{start_time}}' as timestamp) and block_time < cast('{{end_time}}' as timestamp)
```

Comment:
should we invoke the block range query here so that we are consistent throughout?

Reply:
Good question.

Being consistent would be nice. It would also mean that the imbalances query would need to be updated/could not be used.

I would prefer to move towards time based accounting periods everywhere (i.e. using block_number instead of block_deadline (or rather coalesce(block_number, block_deadline) to handle penalties).

Since slippage is (or rather should in principle be) independent of rewards, having inconsistent time ranges is not such a big deal. We have a similar time based accounting for quote rewards. It is arguably more annoying for quote rewards as part of rewards.

Reply:
Ok will continue reviewing but then i would actually suggest we move everything to block numbers and only have the top-level query use dates

Reply:
I think this is not possible, unfortunately. Query parameters in queries you call from within another query cannot be computed values. One can easily forward time ranges as they are not computed but parameters of the calling query. Passing block ranges would require computing them from dates.

Not sure if there is another way.